lib/ex_json_schema/validator/format.ex

```diff
@@ -111,6 +111,7 @@ defmodule ExJsonSchema.Validator.Format do
     case apply(mod, fun, [format, data]) do
       true -> []
       false -> [%Error{error: %Error.Format{expected: format}}]
+      {false, custom_validation_error_format_name} -> [%Error{error: %Error.Format{expected: custom_validation_error_format_name}}]
```

Comment:
Maybe it's a bit more flexible to have the formatter return `{false, %Error.Format{expected: "something"}}` instead? It would even allow you to return your own error struct that implements `String.Chars` to get a completely custom error message.

Reply:
I changed according to your suggestion. Hope this is what you had in mind.